python/ray/serve/api.py

```diff
@@ -571,3 +577,102 @@ def delete(name: str, _blocking: bool = True):
     """
     client = get_global_client()
     client.delete_apps([name], blocking=_blocking)
+
+
+@PublicAPI(stability="alpha")
+def multiplexed(func=None, num_models_per_replica: int = 0):
+    """Coverts a function or method to a multiplexed function.
+
+    The function can be standalone function or a method of a class. The
+    function must have at least one argument with type `str` and the argument
+    must be the first argument. The `str` argument is the model id of the model to
+    be loaded. The function can have other arguments and keyword arguments.
```

Comment:
It seems a little bit dangerous to me to allow other arguments and kwargs... if the resulting model is dependent on those args, then effectively our caching does not work. I would suggest we don't allow this.